lang/cs/Org.Apache.REEF.Wake/Remote/Impl/RemoteEventStreamingCodec.cs

```diff
@@ -30,65 +31,33 @@ namespace Org.Apache.REEF.Wake.Remote.Impl
     /// Writable remote event class
     /// </summary>
     /// <typeparam name="T">Type of remote event message. It is assumed that T implements IWritable</typeparam>
-    [Obsolete("Need to remove Iwritable and use IstreamingCodec. Please see Jira REEF-295 ", false)]
-    internal sealed class WritableRemoteEvent<T> : IWritableRemoteEvent<T> where T : IWritable
+    internal sealed class RemoteEventStreamingCodec<T> : IStreamingCodec<IRemoteEvent<T>>
     {
-        private readonly IInjector _injector;
+        private readonly IStreamingCodec<T> _codec;
 
-        /// <summary>
-        /// Creates the Remote Event
-        /// </summary>
-        /// <param name="localEndpoint">Local Address</param>
-        /// <param name="remoteEndpoint">Remote Address</param>
-        /// <param name="value">Actual message</param>
-        public WritableRemoteEvent(IPEndPoint localEndpoint, IPEndPoint remoteEndpoint, T value)
+        public RemoteEventStreamingCodec(IStreamingCodec<T> codec)
         {
-            LocalEndPoint = localEndpoint;
-            RemoteEndPoint = remoteEndpoint;
-            Value = value;
-        }
-
-        /// <summary>
-        /// Creates empty Remote Event
-        /// </summary>
-        [Inject]
-        public WritableRemoteEvent(IInjector injector)
-        {
-            _injector = injector;
-        }
-
-        /// <summary>
-        /// Local Address
-        /// </summary>
-        public IPEndPoint LocalEndPoint { get; set; }
-
-        /// <summary>
-        /// Remote Address
-        /// </summary>
-        public IPEndPoint RemoteEndPoint { get; set; }
-
-        /// <summary>
-        /// The actual message
-        /// </summary>
-        public T Value { get; set; }
-
+            _codec = codec;
+        } 
+        
         /// <summary>
         /// Read the class fields.
         /// </summary>
         /// <param name="reader">The reader from which to read </param>
-        public void Read(IDataReader reader)
+        /// <returns>The remote event</returns>
+        public IRemoteEvent<T> Read(IDataReader reader)
         {
-            Value = (T)_injector.ForkInjector().GetInstance(typeof(T));
-            Value.Read(reader);         
+            return new RemoteEvent<T>(null, null, _codec.Read(reader));
```

Comment:
These fields are filled in the ObserverContainer which gets it from the underlying link. My guess is this was done since this information is available locally and there is no need to write it as part of message.